packages/flutter/lib/src/widgets/overlay.dart

```diff
@@ -543,6 +543,55 @@ class OverlayState extends State<Overlay> with TickerProviderStateMixin {
     return _entries.length;
   }
 
+  bool _debugCanInsertEntry(OverlayEntry entry) {
+    final List<DiagnosticsNode> operandsInformation = <DiagnosticsNode>[
```

Comment:
probably wrap the whose thing in assert as well, `_debug` prefix should be noop if call directly in debug mode. Not sure if there is a style guide for this, but that is what other _debug method behavior else where in the framework.

Reply:
Sounds like https://github.com/dart-lang/sdk/issues/28113 ? I think I can try adding an annotation (e.g., `@debugAssert`) and prevent it from being called outside of an `assert` in analyze.dart?

Reply:
this is new to me, if it works, i fine with it